src/modules/site/base/views/auth/auth.py

```diff
@@ -20,21 +20,42 @@
                             get_jwt_identity,
                             jwt_required,
                             assign_access_refresh_tokens,
-                            unset_jwt)
+                            unset_jwt,
+                            decode_token,
+                            check_if_token_revoked)
 
 from caendr.models.datastore import User
 from caendr.services.cloud.secret import get_secret
 
 
 PASSWORD_PEPPER = get_secret('PASSWORD_PEPPER')
-auth_bp = Blueprint('auth',
-                    __name__,
-                    template_folder='templates')
+auth_bp = Blueprint('auth', __name__, template_folder='templates')
 
 @auth_bp.route('/')
 def auth():
   return redirect(url_for('auth.choose_login'))
 
+@auth_bp.route('/token')
+def token():
```

Comment:
My thinking for this route was 3 checks that short circuit and re-route to basic login:

1. Check for token existence, validity and expiration
2. Check if token was revoked
3. Check that user associated with token exists

Do you think we can/should combine two of these checks?

When running the 'decode_token()' a couple callbacks can be triggered if the token is expired or invalid which then redirects to login. Those callbacks don't include a check if the token is revoked, that has to be called manually since this route doesn't include @jwt_required.